Sources/UtilityBeltData/CoreData/CoreDataManager.swift

```diff
@@ -0,0 +1,157 @@
+
+    // MARK: - Methods
+
+    // MARK: Initializers
+
+    /// Initializes a new CoreDataManger with no persistent container.
+    /// This is only used for the static default instance of this class.
+    /// All operations on the default instance will fail until the default container is set.
+    private init() {}
+
+    /// Initializes a new CoreDataManager.
+    /// - Parameter container: The persistent container to use for all operations.
+    public required init(container: NSPersistentContainer) {
+        self.persistentContainer = container
+    }
+
+    // MARK: Count
+
+    /// Returns the count of a given managed object.
+    /// - Parameter type: The type of managed object to count.
+    public func count<T: NSManagedObject>(of type: T.Type,
+                                          with predicate: NSPredicate? = nil) throws -> Int {
+        // TODO: Figure out why T.fetchRequest() doesn't work here
+        let fetchRequest = NSFetchRequest<T>(entityName: String(describing: T.self))
+        fetchRequest.includesPropertyValues = false
+        fetchRequest.includesSubentities = false
+        fetchRequest.predicate = predicate
+
+        let count = try self.managedContext().count(for: fetchRequest)
+
+        // If the fetched objects were not found, ensure that we return 0
+        return (count != NSNotFound) ? count : 0
+    }
+
+    // MARK: Create
+
+    /// Creates a new instance of a managed object subclass.
+    /// - Parameter type: The managed object subclass type to create.
+    /// - Parameter context: The managed object context to create the object in. If nil, uses the current default context.
+    public func newInstance<T: NSManagedObject>(of type: T.Type, in context: NSManagedObjectContext? = nil) throws -> T? {
+        let context = try (context ?? self.managedContext())
+
+        return NSEntityDescription.insertNewObject(forEntityName: String(describing: T.self), into: context) as? T
+    }
+
+    // MARK: Delete
+
+    /// Deletes a single object.
+    /// - Parameter object: The object to delete.
+    public func delete<T: NSManagedObject>(_ object: T) throws {
+        let context = try (object.managedObjectContext ?? self.managedContext())
+
+        context.delete(object)
+        try context.save()
+    }
+
+    /// Deletes all objects of a given entity type.
+    /// - Parameter type: The entity type to batch delete.
+    ///
+    /// Batch delete requests are only compatible with the SQLite store type.
+    public func deleteAll<T: NSManagedObject>(of type: T.Type,
+                                              with predicate: NSPredicate? = nil) throws {
+        let fetchRequest = NSFetchRequest<NSFetchRequestResult>(entityName: String(describing: T.self))
+        fetchRequest.predicate = predicate
+
+        let deleteRequest = NSBatchDeleteRequest(fetchRequest: fetchRequest)
+
+        try self.managedContext().executeAndMergeChanges(using: deleteRequest)
```

Comment:
should the context be optionally passed in and default to `self.managedContext`?